ui/src/pageLayout/containers/Nav.tsx

```diff
@@ -58,314 +54,141 @@ class SideNav extends PureComponent<Props, State> {
+                      <TreeNav.SubItem
+                        key={menuItem.id}
+                        id={menuItem.id}
+                        testID={menuItem.testID}
+                        active={getNavItemActivation(
+                          [menuItem.id],
+                          location.pathname
+                        )}
+                        label={menuItem.label}
+                        linkElement={className => (
+                          <Link className={className} to={menuItem.link} />
+                        )}
+                      />
+                    )
+
+                    if (menuItem.cloudExclude) {
+                      navSubItemElement = (
+                        <CloudExclude key={menuItem.id}>
+                          {navSubItemElement}
+                        </CloudExclude>
+                      )
+                    }
+
+                    if (menuItem.cloudOnly) {
+                      navSubItemElement = (
+                        <CloudOnly key={menuItem.id}>
+                          {navSubItemElement}
+                        </CloudOnly>
+                      )
+                    }
+
+                    if (menuItem.featureFlag) {
+                      navSubItemElement = (
+                        <FeatureFlag
+                          key={menuItem.id}
+                          name={menuItem.featureFlag}
+                        >
+                          {navSubItemElement}
+                        </FeatureFlag>
+                      )
+                    }
+
+                    return navSubItemElement
+                  })}
+                </TreeNav.SubMenu>
+              ) : null}
+            </TreeNav.Item>
+          )
+
+          if (item.cloudExclude) {
+            navItemElement = (
+              <CloudExclude key={item.id}>{navItemElement}</CloudExclude>
+            )
+          }
+
+          if (item.cloudOnly) {
+            navItemElement = (
+              <CloudOnly key={item.id}>{navItemElement}</CloudOnly>
+            )
+          }
+
+          if (item.featureFlag) {
+            navItemElement = (
+              <FeatureFlag key={item.id} name={item.featureFlag}>
+                {navItemElement}
+              </FeatureFlag>
+            )
+          }
+
+          return navItemElement
+        })}
+      </TreeNav>
     )
   }
+}
 
-  private showOrganizationsView = (): void => {
-    this.setState({isShowingOrganizations: true})
-  }
-
-  private closeOrganizationsView = (): void => {
-    this.setState({isShowingOrganizations: false})
-  }
+const mdtp: DispatchProps = {
+  handleSetNavBarState: setNavBarState,
 }
 
 const mstp = (state: AppState): StateProps => {
   const isHidden = state.app.ephemeral.inPresentationMode
-  const orgs = getAll<Organization>(state, ResourceType.Orgs)
-  const org = getOrg(state)
-  const {me} = state
+  const navBarState = state.app.persisted.navBarState
```

Comment:
i'd love to see this use `get` just so we don't end up with an error at some point saying `can't read property "persisted" of undefined`

Reply:
Are you referring to `get` from lodash or something else?

Reply:
Yes please